### PR TITLE
enhance: add more metrics

### LINKED ISSUE #31272: [Enhancement]: add more metrics
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

I want to add more metrics, which are:

- Slow query count, which the duration considered as slow can be configurable;
- Number of deleted entities;
- Number of entities imported;
- Number of entities per collection;
- Number of loaded entities per collection;
- Number of indexed entities;
- Number of indexed entities, per collection, per index and whether it's a vetor index;
- Quota states (LongTimeTickDelay, MemoryExhuasted, DiskQuotaExhuasted) per database;

### Why is this needed?

Make the observability of Milvus better.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by longjiquan**
/kind improvement
fix: #31272 

This pr add more metrics, which are:
- Slow query count, which the duration considered as slow can be configurable;
- Number of deleted entities;
- Number of entities imported;
- Number of entities per collection;
- Number of loaded entities per collection;
- Number of indexed entities;
- Number of indexed entities, per collection, per index and whether it's a vetor index;
- Quota states (LongTimeTickDelay, MemoryExhuasted, DiskQuotaExhuasted) per database;

### CODE REVIEW

File: internal/datacoord/import_scheduler.go [hunk 1]
```diff
@@ -286,6 +289,11 @@ func (s *importScheduler) processInProgressImport(task ImportTask) {
 			log.Warn("update import segment rows failed", WrapTaskLog(task, zap.Error(err))...)
 			return
 		}
+		metrics.DataCoordBulkVectors.WithLabelValues(
+			strconv.FormatInt(paramtable.GetNodeID(), 10),
```
Comment (bigsheeper) [inline internal/datacoord/import_scheduler.go:293]: Why is the nodeID needed?

File: internal/datacoord/import_scheduler.go [hunk 2]
```diff
@@ -286,6 +289,11 @@ func (s *importScheduler) processInProgressImport(task ImportTask) {
...
+			strconv.FormatInt(task.GetCollectionID(), 10),
+			strconv.FormatInt(task.GetJobID(), 10),
```
Comment (bigsheeper) [inline internal/datacoord/import_scheduler.go:295]: Why is the jobID needed?

File: internal/datacoord/import_scheduler.go (hunk 1 above)
Comment (longjiquan) [reply internal/datacoord/import_scheduler.go:293]: This metric is reported by datanode in the first design, so nodeID is needed. Forgot to remove this label when I move this to datacoord.

File: internal/datacoord/import_scheduler.go (hunk 2 above)
Comment (longjiquan) [reply internal/datacoord/import_scheduler.go:295]: @czs007

File: internal/proxy/impl.go [hunk 3]
```diff
@@ -2674,8 +2674,12 @@ func (node *Proxy) Search(ctx context.Context, request *milvuspb.SearchRequest)
 
 	defer func() {
 		span := tr.ElapseSpan()
-		if span >= SlowReadSpan {
+		if float64(span.Milliseconds()) >= paramtable.Get().ProxyCfg.SlowQuerySpanInSeconds.GetAsFloat()*1000 {
```
Comment (bigsheeper) [inline internal/proxy/impl.go:2677]: paramtable.Get().ProxyCfg.SlowQuerySpanInSeconds.GetAsDuration()

File: internal/proxy/impl.go [hunk 4]
```diff
@@ -2814,8 +2818,12 @@ func (node *Proxy) HybridSearch(ctx context.Context, request *milvuspb.HybridSea
...
```
Comment (bigsheeper) [inline internal/proxy/impl.go:2821]: GetAsDuration()

File: internal/proxy/impl.go [hunk 5]
```diff
@@ -3069,14 +3077,18 @@ func (node *Proxy) query(ctx context.Context, qt *queryTask) (*milvuspb.QueryRes
...
```
Comment (bigsheeper) [inline internal/proxy/impl.go:3080]: GetAsDuration()

File: internal/querynodev2/metrics_info.go [hunk 6]
```diff
@@ -154,6 +154,21 @@ func getQuotaMetrics(node *QueryNode) (*metricsinfo.QueryNodeQuotaMetrics, error
...
 
+func getCollectionMetrics(node *QueryNode) (*metricsinfo.QueryNodeCollectionMetrics, error) {
+	allSegments := node.manager.Segment.GetBy()
+	ret := &metricsinfo.QueryNodeCollectionMetrics{
+		Collections: make(map[int64]int64),
+	}
+	for _, segment := range allSegments {
+		collectionID := segment.Collection()
+		if _, ok := ret.Collections[collectionID]; !ok {
+			ret.Collections[collectionID] = 0
```
Comment (bigsheeper) [inline internal/querynodev2/metrics_info.go:165]: Unnecessary initialization

File: internal/rootcoord/quota_center.go [hunk 7]
```diff
@@ -177,6 +178,61 @@ func (q *QuotaCenter) clearMetrics() {
 	q.proxyMetrics = make(map[UniqueID]*metricsinfo.ProxyQuotaMetrics, 0)
 }
 
+func updateNumEntitiesLoaded(current map[int64]int64, qn *metricsinfo.QueryNodeCollectionMetrics) map[int64]int64 {
+	for collectionID, rowNum := range qn.Collections {
+		if _, ok := current[collectionID]; !ok {
+			current[collectionID] = 0
```
Comment (bigsheeper) [inline internal/rootcoord/quota_center.go:184]: Unnecessary initialization

File: internal/rootcoord/quota_center.go [hunk 8]
```diff
@@ -177,6 +178,61 @@ func (q *QuotaCenter) clearMetrics() {
...
+			continue
+		}
+		metrics.RootCoordNumEntities.WithLabelValues(info.Name, metrics.TotalLabel).Set(float64(collection.NumEntitiesTotal))
+		fields := lo.KeyBy(info.Fields, func(v *model.Field) int64 { return v.FieldID })
+		for _, indexInfo := range collection.IndexInfo {
+			if _, ok := fields[indexInfo.FieldID]; !ok {
+				log.Warn("field id not found, ignore to report indexed num entities",
+					zap.Int64("collection", collectionID),
+					zap.Int64("field", indexInfo.FieldID),
+				)
```
Comment (bigsheeper) [inline internal/rootcoord/quota_center.go:225]: continue

File: internal/rootcoord/quota_center.go [hunk 9]
```diff
@@ -266,6 +331,8 @@ func (q *QuotaCenter) syncMetrics() error {
 		return err
 	}
 
+	q.reportNumEntitiesLoaded(numEntitiesLoaded)
```
Comment (bigsheeper) [inline internal/rootcoord/quota_center.go:334]: like datacoord, report after GetMetrics?

File: pkg/util/metricsinfo/metrics_info.go [hunk 10]
```diff
@@ -94,11 +94,16 @@ type QueryNodeConfiguration struct {
 	SimdType string `json:"simd_type"`
 }
 
+type QueryNodeCollectionMetrics struct {
+	Collections map[int64]int64
```
Comment (bigsheeper) [inline pkg/util/metricsinfo/metrics_info.go:98]: Unclear naming, better to rename to collectionRows or anything else

File: internal/rootcoord/quota_center.go (hunk 9 above)
Comment (longjiquan) [reply internal/rootcoord/quota_center.go:334]: ok

File: pkg/util/metricsinfo/metrics_info.go (hunk 10 above)
Comment (longjiquan) [reply pkg/util/metricsinfo/metrics_info.go:98]: ok

**Comment by longjiquan** [comment]
rerun ut

**Comment by czs007** [comment]
@longjiquan remember to pick to 2.4 branch

**Comment by czs007** [comment]
rerun ut

**Pull request merged**